connectors/src/connectors/confluence/index.ts

```diff
@@ -395,6 +400,79 @@ export async function retrieveConfluenceObjectsTitles(
   return new Ok(titles);
 }
 
+export async function retrieveConfluenceContentNodes(
+  connectorId: ModelId,
+  internalIds: string[]
+): Promise<Result<ConnectorNode[], Error>> {
+  const connectorState = await ConfluenceConfiguration.findOne({
+    where: {
+      connectorId,
+    },
+  });
+  if (!connectorState) {
+    return new Err(new Error("Confluence configuration not found"));
+  }
+
+  const spaceIds: string[] = [];
+  const pageIds: string[] = [];
+
+  internalIds.forEach((internalId) => {
+    if (isConfluenceInternalSpaceId(internalId)) {
+      spaceIds.push(getIdFromConfluenceInternalId(internalId));
+    } else if (isConfluenceInternalPageId(internalId)) {
+      pageIds.push(getIdFromConfluenceInternalId(internalId));
+    }
+  });
+
+  const [confluenceSpaces, confluencePages] = await Promise.all([
+    ConfluenceSpace.findAll({
+      where: {
+        connectorId: connectorId,
+        spaceId: spaceIds,
+      },
+    }),
+    ConfluencePage.findAll({
+      where: {
+        connectorId: connectorId,
+        pageId: pageIds,
+      },
+    }),
+  ]);
+
+  const spaceContentNodes: ConnectorNode[] = confluenceSpaces.map((space) => {
+    return createConnectorNodeFromSpace(space, connectorState.url, "read", {
+      isExpandable: true,
```

Comment:
`nit` What if the space does not have any pages?

Reply:
The line will be expandable with nothing inside it, which looks legit as at least you know that the Space you selected is empty